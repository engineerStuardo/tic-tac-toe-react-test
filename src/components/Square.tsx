import { SquareProps } from './interfaces';

const Square = ({ value, onClick }: SquareProps) => {
  const style = value ? `squares ${value}` : `squares`;

  return (
    <button className={style} onClick={onClick}>
      {value}
    </button>
  );
};

export default Square;
