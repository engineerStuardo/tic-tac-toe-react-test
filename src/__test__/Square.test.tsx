import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';

import Square from '../components/Square';

const value = 'X';
const handleClick = jest.fn();

describe('Square component', () => {
  test('Should match a snapshot', () => {
    const { container } = render(
      <Square value={value} onClick={handleClick} />
    );
    expect(container).toMatchSnapshot();
  });

  test('Should display value text in button', () => {
    render(<Square value={value} onClick={handleClick} />);
    expect(screen.getByText('X')).toBeTruthy();
  });

  test('Calls onClick prop when clicked', () => {
    render(<Square onClick={handleClick} value={value} />);
    fireEvent.click(screen.getByText(/X/i));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
