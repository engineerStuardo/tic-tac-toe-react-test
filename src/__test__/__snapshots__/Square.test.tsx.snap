// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square component Should match a snapshot 1`] = `
<div>
  <button
    class="squares X"
  >
    X
  </button>
</div>
`;
